### PR TITLE
router_api: bind to / should points to an empty upstream

### PR DESCRIPTION
**Pull request by morpheu**
Change bind to support empty upstream instead default to destination hostname -
which is invalid when using it in router mode.

### CODE REVIEW

File: rpaas/manager.py [hunk 1]
```diff
@@ -171,7 +171,10 @@ def bind(self, name, app_host):
                 return
             if bound_host is not None:
                 raise BindError("This service can only be bound to one application.")
-        self.consul_manager.write_location(name, "/", destination=app_host)
+        empty_upstream = False
+        if router_mode:
+            empty_upstream = True
+        self.consul_manager.write_location(name, "/", destination=app_host, empty_upstream=empty_upstream)
```
Comment (cezarsa) [inline rpaas/manager.py]: Wouldn't using `empty_upstream=router_mode` here be the same?

File: rpaas/consul_manager.py [hunk 2]
```diff
@@ -152,18 +155,25 @@ def remove_lua(self, instance_name, lua_module_name, lua_module_type):
 
     def add_server_upstream(self, instance_name, upstream_name, server):
         servers = self.list_upstream(instance_name, upstream_name)
+        if not server:
```
Comment (cezarsa) [inline rpaas/consul_manager.py]: Maybe move this check to before calling `list_upstream`?

File: rpaas/consul_manager.py [hunk 3]
```diff
@@ -152,18 +155,25 @@ def remove_lua(self, instance_name, lua_module_name, lua_module_type):
...
+            return
         if isinstance(server, list):
+            for idx, _ in enumerate(server):
+                server[idx] = ":".join(map(str, filter(None, self._host_from_destination(server[idx]))))
```
Comment (cezarsa) [inline rpaas/consul_manager.py]: What's `filter(None, ...` trying to achieve here? Isn't this the same as simply using the tuple returned by `_host_from_destination`?

Comment (morpheu) [reply rpaas/consul_manager.py]: In some situations destinations doesn't contain port.

File: rpaas/manager.py (hunk 1 above)
Comment (morpheu) [reply rpaas/manager.py]: Ouch!

File: rpaas/consul_manager.py (hunk 3 above)
Comment (cezarsa) [reply rpaas/consul_manager.py]: ah, okay

**Pull request merged**